src/backtrace/dbghelp.rs

```diff
@@ -167,11 +167,21 @@ pub unsafe fn trace(cb: &mut dyn FnMut(&super::Frame) -> bool) {
         Some(StackWalkEx) => {
             let mut stack_frame_ex: STACKFRAME_EX = mem::zeroed();
             stack_frame_ex.StackFrameSize = mem::size_of::<STACKFRAME_EX>() as DWORD;
-            stack_frame_ex.AddrPC.Offset = context.0.Eip as u64;
+
+            cfg_if::cfg_if! {
+                if #[cfg(target_arch = "x86")] {
+                        stack_frame_ex.AddrPC.Offset = context.0.Eip as u64;
+                        stack_frame_ex.AddrStack.Offset = context.0.Esp as u64;
+                        stack_frame_ex.AddrFrame.Offset = context.0.Ebp as u64;
+                } else {
+                        stack_frame_ex.AddrPC.Offset = context.0.Pc as u64;
+                        stack_frame_ex.AddrStack.Offset = context.0.Sp as u64;
+                        stack_frame_ex.AddrFrame.Offset = context.0.R11 as u64;
+                }
+            }
+
```

Comment:
Rather than repeating this `cfg_if` block, it'd be better to add a platform specific implementations on `MyContext`, similar to https://github.com/rust-lang/backtrace-rs/blob/e9da96eb452aa65e79e2342be700544afe509440/src/backtrace/dbghelp.rs#L76-L81.